prov/usnic/src/usdf_cm.c

```diff
@@ -563,26 +570,89 @@ usdf_cm_msg_connect(struct fid_ep *fep, const void *addr,
 	return ret;
 }
 
+/* A wrapper to core function to translate string address to
+ * sockaddr_in type. We are expecting a NULL sockaddr_in**.
+ * The core function will allocated it for us. The caller HAS TO FREE it.
+ */
+int usdf_str_toaddr(const char *str, struct sockaddr_in **outaddr)
+{
+	uint32_t type;
+	size_t size;
+	int ret;
+
+	type = FI_SOCKADDR_IN;
+
+	/* call the core function. The core always allocate the addr for us. */
+	ret = ofi_str_toaddr(str, &type, (void **)outaddr, &size);
+
+#if ENABLE_DEBUG
+	char outstr[USDF_ADDR_STR_LEN];
+
+	/* verify the output in debug. */
```

Comment:
oh well `str_toaddr` is also just calling the core. hmm.
I guess I will add debug in both functions then.